packages/smb-client/src/client/broker-client.class.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@peque/di';
+import {
+  BrokerSocket,
+  Command,
+  CommandParser,
+  EventService,
+  IMessageCommand,
+  ISubscribeCommand,
+  IUnsubscribeCommand,
+  IWelcomeCommand,
+  SocketService,
+} from '@peque/smb-commons';
+
+import { MessageCommand, PublishCommand, SubscribeCommand, UnsubscribeCommand } from '../commands';
+import { IBrokerClientOptions, Listener } from '../models';
+import { BrokerConnectionTimeoutException } from '../models';
+import { SubscribeListenerService } from '../services';
+
+@Injectable()
+export class BrokerClient {
+  #connectionTimeout: NodeJS.Timer;
+  #socket: BrokerSocket;
+
+  constructor(
+    private sockets: SocketService,
+    private events: EventService,
+    private commandParser: CommandParser,
+    private command: Command,
+    private subscriptions: SubscribeListenerService,
+    private messageCommand: MessageCommand,
+    private publishMessage: PublishCommand,
+    private subscribeCommand: SubscribeCommand,
+    private unsubscribeCommand: UnsubscribeCommand,
+  ) {
+    this.command.init([messageCommand, publishMessage, subscribeCommand, unsubscribeCommand]);
+  }
+
+  async connect(options?: IBrokerClientOptions): Promise<string> {
+    return new Promise((resolve, reject) => {
+      this.#socket = new BrokerSocket();
+
+      const welcomeMsg = (welcome: IWelcomeCommand) => {
+        this.events.remove('welcome', welcomeMsg);
+        this.#socket.id = welcome.socketId;
+        this.sockets.set(this.#socket);
+        resolve(welcome.socketId);
+      };
+
+      this.events.on('welcome', welcomeMsg);
+
+      this.#socket.connect({ port: options?.port || 8021, host: options?.host || '127.0.0.1' });
+      this.#socket.on('data', (data) => this.events.next('incomingCommand', data));
+
+      this.#connectionTimeout = setTimeout(() => {
+        this.#socket.removeAllListeners('data');
+        clearTimeout(this.#connectionTimeout);
+        reject(new BrokerConnectionTimeoutException());
```

Comment:
Could we rename this to use the word `Error`?
```suggestion
        reject(new BrokerConnectionTimeoutError());
```